src/ansys/simai/core/data/training_data.py

```diff
@@ -84,28 +84,33 @@ def get_subset(self, project: Identifiable["Project"]) -> Optional[SubsetEnum]:
                 the :class:`~.projects.Project` object for, or its ID.
 
         Returns:
-            SubsetEnum of the subset that the training data belongs to in the given project.
-                (e.g. <SubsetEnum.VALIDATION: 'Validation'>)
+            None if subset is unassigned or SubsetEnum of the subset that the training data belongs to in the given project.
+                (e.g. <SubsetEnum.TEST: 'Test'>)
         """
         project_id = get_id_from_identifiable(project, default=self._client._current_project)
         subset_value = self._client._api.get_training_data_subset(project_id, self.id).get("subset")
         return SubsetEnum(subset_value) if subset_value else None
 
-    def assign_subset(self, project: Identifiable["Project"], subset: SubsetEnum) -> None:
+    def assign_subset(self, project: Identifiable["Project"], subset: Optional[SubsetEnum]) -> None:
         """Assign the training data subset in relation to a given project.
 
         Args:
             project: ID or :class:`model <.projects.Project>`
-            subset: SubsetEnum attribute (e.g. SubsetEnum.TRAINING) or string value (e.g. "Training").
-                Available options: (Training, Validation, Test, Ignored)
+            subset: None to unassign or SubsetEnum attribute (e.g. SubsetEnum.TRAINING) or string value (e.g. "Training").
+                Available options: (Training, Test)
 
         Returns:
             None
         """
-        if subset not in SubsetEnum.__members__.values():
-            raise InvalidArguments("Must be one of: Ignored, Training, Test, Validation.")
+        if subset is not None and subset not in SubsetEnum.__members__.values():
+            raise InvalidArguments("Must be None or one of: 'Training', 'Test'.")
+
         project_id = get_id_from_identifiable(project, default=self._client._current_project)
-        self._client._api.put_training_data_subset(project_id, self.id, subset)
+        if subset is None:
+            self._client._api.remove_training_data_from_project(self.id, project_id)
+            self._client._api.add_training_data_to_project(self.id, project_id)
```

Comment:
That's a hack, maybe `put_training_data_subset` should allow `None` so we don't need this